### DISCUSSION TITLE
The difference between seleniumbase and selenium

### DISCUSSION THREAD

**Discussion by Stevenphil** [seleniumbase/SeleniumBase#1156] [Q&A] [answered]
First of all thank you for your work, I want to know what is the difference between seleniumbase and selenium

**Comment by mdmintz** [comment, marked as answer]
Hi @Stevenphil, SeleniumBase is a complete Web-UI testing framework that merges the unit-testing abilities of pytest with the Python version of Selenium. SeleniumBase also adds a lot of functionality that isn't found in either pytest or Selenium, such as new pytest command-line options, better reliability when calling Selenium methods, and advanced tools such as the Dashboard, the Recorder, screenshots for failing tests, and more. Tests are given structure, such as automatic browser launching and closing at the start and end of tests. Methods are simplified, which allow for shorter tests that are easier to understand. See https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/features_list.md for a much longer list of features. There are also 100s of examples in https://github.com/seleniumbase/SeleniumBase/tree/master/examples, which help newcomers learn by example. For the full list of command-line options, see https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/customizing_test_runs.md. For the full API (test methods), see https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/method_summary.md.

**Reply by Stevenphil** [reply]
Wow, it is too powerful, I will study it carefully，thank you very much！
